src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import duke.commands.Command;
+import duke.exceptions.DukeException;
+import duke.exceptions.NoSuchCommandException;
+import duke.exceptions.NoSuchTaskException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Duke {
+
+    private Storage storage;
+    private Tasklist tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = storage.fileToTasklist();
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new Tasklist();
+        }
+    }
+
+    public void run() {
+        ui.showWelcomeMessage();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                //#ui.showLine(); // show the divider line ("_______")
```

Comment:
It's free to delete lines which are commented out to make the code clear, since you can still review the old version through source tree or git if u think it's better to change back.